### ISSUE TITLE
utils.sh(truncate_large_logs): cp: cannot stat ‘’: No such file or directory

### ISSUE THREAD

**Issue by php-coder** [openshift/origin#6788]
```
$ ./test/extended/core.sh --ginkgo.focus="mysql"
...
[INFO] Removing k8s docker containers
cp: cannot stat ‘’: No such file or directory
./test/extended/../../hack/util.sh: line 500: : No such file or directory
./test/extended/../../hack/util.sh: line 501: : No such file or directory
rm: cannot remove ‘.tmp’: No such file or directory
[INFO] Cleanup complete
[INFO] Exiting
```

Maybe it's the same as #6761 but I'm not sure.

**Comment by stevekuznetsov** [comment]
@php-coder Are you getting this on current HEAD?

**Comment by php-coder** [comment]
Yes, on the 4e1247 commit to be precise.

**Comment by liggitt** [comment]
@stevekuznetsov the setup functions make a lot of assumptions about a full build having been run, but not all the test scripts that delegate to them need a full build. I see errors about the `openshift` binary not existing when trying to determine IP addresses for the node for tests that don't even start a node... probably need to wrap more conditionals around some of the checks if we're going to call these setup/teardown functions everywhere

**Comment by stevekuznetsov** [comment]
@liggitt Agreed, I'm currently breaking up the cleanup scripts into smaller pieces so we can choose when we do what (i.e. kill all processes versus stop docker image, etc). I will include those conditionals there like you say.

**Comment by deads2k** [comment]
> @liggitt Agreed, I'm currently breaking up the cleanup scripts into smaller pieces so we can choose when we do what (i.e. kill all processes versus stop docker image, etc). I will include those conditionals there like you say.

@stevekuznetsov what actually happens in the scripts is that everyone will call the biggest cleanup they can (and  you kind of want them to) and that cleanup should just gracefully handling cleaning up things that aren't present.  It already needs to do that to properly handle failures at any point.

Short version: don't make a bunch of little cleanup functions, make the big one work properly.

**Comment by stevekuznetsov** [comment]
@deads2k and when we run `test-end-to-end-docker`, we write another version of the monolithic cleanup function that does 99.99% the same thing except it stops the `origin` Docker container instead of killing the OpenShift process? No, I don't think so. Monolithic cleanup functions will continue to exit, but they will delegate to smaller pluggable modules. Functions that start a service (think `start_os_server` or the new system logging func) will automatically install the cleanup. Any further tweaks will have to be done manually.

**Comment by liggitt** [comment]
More fun things in the exit hooks.

Seen when running the integration tests, which don't require docker:
```
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

Seen when running the integration tests, which launch a secured API server (this one actually hangs the console at a login prompt):
```
[INFO] Dumping all resources to /tmp/openshift/test-integration//logs/export_all.json
Server [https://localhost:8443]: 
```

**Comment by liggitt** [comment]
Seen after running hack/test-integration-docker.sh successfully:
```
[INFO] Dumping all resources to /tmp/openshift/test-integration//logs/export_all.json
The connection to the server 127.0.0.1:12000 was refused - did you specify the right host or port?
The connection to the server 127.0.0.1:12000 was refused - did you specify the right host or port?
```

**Comment by stevekuznetsov** [comment]
@liggitt I think most of the issues we are seeing right now are due to everyone just using `cleanup_openshift` which is a grab-bag of things originally intended for use in extended tests where a node is running. Most of these issues will therefore be fixed by my changes as we can enable cleanup on a much more fine-grained level. I added a `$ADMIN_KUBECONFIG` check to the `export_all` cleanup function yesterday, as I noticed it could hang either on the insecure verify step or the login step without it, which could happen on a premature error as some non-trivial amount of work happens before we can set `$ADMIN_KUBECONFIG`.
